src/globus_sdk/services/auth/_common.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import annotations
+
+from globus_sdk._types import ScopeCollectionType
+from globus_sdk.exc import GlobusSDKUsageError
+from globus_sdk.exc.warnings import warn_deprecated
+from globus_sdk.scopes import AuthScopes, MutableScope, TransferScopes
+
+_DEFAULT_REQUESTED_SCOPES = (
+    AuthScopes.openid,
+    AuthScopes.profile,
+    AuthScopes.email,
+    TransferScopes.all,
+)
+
+
+def stringify_requested_scopes(requested_scopes: ScopeCollectionType | None) -> str:
+    if requested_scopes is None:
+        warn_deprecated(
+            "`requested_scopes` was not specified or was given as `None`. "
+            "A default set of scopes will be used, but this behavior is deprecated. "
+            "Specify an explicit set of scopes instead.",
+            stacklevel=3,
+        )
+        requested_scopes = _DEFAULT_REQUESTED_SCOPES
+
+    requested_scopes_string: str = MutableScope.scopes2str(requested_scopes)
+    if requested_scopes_string == "":
+        raise GlobusSDKUsageError(
+            "requested_scopes cannot be the empty string or empty collection"
+        )
```

Comment:
Are we 100% there isn't a use case for this?

Feels like the type of thing where a customer will come back asking us to [re-enable spacebar heating](https://xkcd.com/1172/) and I want to be sure there aren't any actual valid usages for this edge case.

Reply:
A user or two could, in theory, ask for us to re-enable support for `requested_scopes=""`, but my feeling is that this has been a bug for as long as we've treated that as valid, and we're fixing it. The incompatible change is that we used to treat `""` or `[]` as meaning the default as well as `None`, but that wasn't really intentional as far as I can tell.

I don't mean that there's no risk that a user is going to be bitten by this, but I'm just not inclined to treat the behavior of `requested_scopes=[]` as covered by our semver "contract" with the user.

To clarify why this is a bug IMO: a user could reasonably expect that `requested_scopes=""` means "request no scopes". Such a usage is never valid, and we silently converted it to valid usage.

They weren't really getting `""` as their scope string. They were getting the defaults. The better way for them to get the defaults is either to pass no scopes or to pass `None` explicitly.

---

This possibility -- that the _computed_ scope string is the empty string -- is actually a newly introduced situation in this change.

The only way I know of to get `""` out of the existing code in `main` is to define an iterable which is empty but which bools as "True". e.g.
```python
class MyList(list):
    def __bool__(self): return True

x = MyList()  # try in an interpreter, `x or "foo"` is []
...

requested_scopes=MyList()
```

At that point, I'm like "yeah, you got me! good job! gold star! go away" 😹

Reply:
Ah yeah, I didn't think correctly about how this was handled previously when writing this comment.

Happy to resolve.